search/routes/api/serialize.py

```diff
@@ -122,31 +127,84 @@ def as_json(document_or_set: Union[DocumentSet, Document],
 class AtomXMLSerializer(BaseSerializer):
     """Atom XML serializer for paper metadata."""
 
-    ATOM = "http://www.w3.org/2005/Atom"
-    OPENSEARCH = "http://a9.com/-/spec/opensearch/1.1/"
-    ARXIV = "http://arxiv.org/schemas/atom"
-    NSMAP = {
-        None: ATOM,
-        "opensearch": OPENSEARCH,
-        "arxiv": ARXIV
-    }
-#     fields = {
-#         'title': '{%s}title' % ATOM,
-#         'id': '{%s}id' % ATOM,
-#         'submitted_date': '{%s}published' % ATOM,
-#         'modified_date': '{%s}updated' % ATOM,
-#         'abstract': '{%s}summary' % ATOM,
-#         ''
-#     }
-#
-#     def __init__(cls, *args, **kwargs) -> None:
-#         super(AtomXMLSerializer, cls).__init__(*args, **kwargs)
-#         cls._root = etree.Element('feed', nsmap=cls.NSMAP)
-#
-#     def transform(cls):
-#         for document in cls.iter_documents():
-#
-#
-#
-#     def __repr__(cls) -> str:
-#         return etree.tostring(cls._root, pretty_print=True)
+    @classmethod
+    def transform_document(cls, fg: FeedGenerator, doc: Document,
+                           query: Optional[APIQuery] = None) -> None:
+        """Select a subset of :class:`Document` properties for public API."""
+        entry = fg.add_entry()
+        entry.id(url_for("api.paper", paper_id=doc['paper_id'],
+                             version=doc['version'], _external=True))
+        entry.title(doc['title'])
+        entry.summary(doc['abstract'])
+        entry.published(doc['submitted_date'])
+        entry.updated(doc['updated_date'])
+        entry.link({'href': url_for("api.paper", paper_id=doc['paper_id'], version=doc['version'], _external=True),
```

Comment:
Looks like the classic API has several link elements, e.g.:

```xml
    <link title="doi" href="http://dx.doi.org/10.1142/S021797921244002X" rel="related"/>
    <link href="http://arxiv.org/abs/1201.4869v1" rel="alternate" type="text/html"/>
    <link title="pdf" href="http://arxiv.org/pdf/1201.4869v1" rel="related" type="application/pdf"/>
```

We should be sure to implement all three of these. 

The link element in the current implementation is a little confusing, because it points at an API endpoint, but has ``"type": "text/html"``.

Reply:
I could could fix this as an f-string with a constant for the "https://arxiv.org/abs/" endpoint. Is there  a better mechanism using `url_for`? 

This will also resolve the issue with implementing the PDF links

Reply:
See https://github.com/arXiv/arxiv-base/blob/master/arxiv/base/urls/__init__.py